editor/src/components/canvas/canvas-strategies/strategies/grid-move-reorder-strategy.ts

```diff
@@ -0,0 +1,117 @@
+import { MetadataUtils } from '../../../../core/model/element-metadata-utils'
+import * as EP from '../../../../core/shared/element-path'
+import { controlsForGridPlaceholders } from '../../controls/grid-controls-for-strategies'
+import type { CanvasStrategyFactory } from '../canvas-strategies'
+import { onlyFitWhenDraggingThisControl } from '../canvas-strategies'
+import type { InteractionCanvasState } from '../canvas-strategy-types'
+import {
+  getTargetPathsFromInteractionTarget,
+  emptyStrategyApplicationResult,
+  strategyApplicationResult,
+} from '../canvas-strategy-types'
+import type { InteractionSession } from '../interaction-state'
+import { getGridElementPinState } from './grid-helpers'
+import { isInfinityRectangle } from '../../../../core/shared/math-utils'
+import {
+  getCommandsAndPatchForGridReorder,
+  getGridTemplates,
+  gridMoveStrategiesExtraCommands,
+} from './grid-move-helpers'
+
+export const gridMoveReorderStrategy: CanvasStrategyFactory = (
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession | null,
+) => {
+  const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+  if (
+    selectedElements.length === 0 ||
```

Comment:
```suggestion
    selectedElements.length !== 1 ||
```
Otherwise this might do something that appears strange if more than one thing is selected.

Reply:
[`5ab990f` (#6606)](https://github.com/concrete-utopia/utopia/pull/6606/commits/5ab990f670cb23ea896fd904d4720913ff230076)